iosxr_iso2vbox.py

```diff
@@ -65,38 +63,295 @@
 import time
 import subprocess
 import getpass
-import shlex
 import argparse
 from argparse import RawDescriptionHelpFormatter
 import re
+import logging
+import pexpect
 
 # Telnet ports used to access IOS XR via socat
-console_port = 65000
-aux_port = 65001
+CONSOLE_PORT = 65000
+AUX_PORT = 65001
 
+logger = logging.getLogger(__name__)
 
-def run(command_string, debug=False):
+
+def set_logging():
+    '''
+    Set basic logging format.
+    '''
+    FORMAT = "[%(filename)s:%(lineno)s - %(funcName)20s() ] %(message)s"
+    logging.basicConfig(format=FORMAT)
+
+
+def run(cmd, hide_error=False, cont_on_error=False):
+    '''
+    Run command to execute CLI and catch errors and display them whether
+    in verbose mode or not.
+
+    Allow the ability to hide errors and also to continue on errors.
+    '''
+    output = subprocess.Popen(cmd,
+                              stdin=subprocess.PIPE,
```

Comment:
Unless you're actually supporting piping input into the subprocess, I don't think you want to set stdin here.